source/grid/grid_generator.cc

```diff
@@ -7250,29 +7363,101 @@ namespace GridGenerator
         // create cells one by one
 
         if (dim == 2)
-          for (const auto &cell : table_2D_cell)
-            add_cell(dim, cell, current_material_id);
+          {
+            const auto current_manifold_id_of_cell =
+              cell.manifold_id(); // get cell-manifold id from current quad cell
+            for (const auto &cell : table_2D_cell)
+              add_cell(dim,
+                       cell,
+                       current_material_id,
+                       current_manifold_id_of_cell); // set cell-manifold id
+
+            // inherit inner manifold (faces)
+            if (current_manifold_id_of_cell != numbers::flat_manifold_id)
+              for (const auto &face : table_2D_inner_faces)
+                // set inner tri-faces according to cell-manifold of quad
+                // element, set invalid b_id, since we do not want to modify
+                // b_id inside
+                add_cell(1,
+                         face,
+                         numbers::internal_face_boundary_id,
+                         current_manifold_id_of_cell);
+          }
         else if (dim == 3)
-          for (const auto &cell : table_3D_cell)
-            add_cell(dim, cell, current_material_id);
+          {
+            const auto current_manifold_id_of_cell = cell.manifold_id();
+            for (const auto &cell : table_3D_cell)
+              add_cell(
+                dim,
+                cell,
+                current_material_id,
+                current_manifold_id_of_cell); // set all cell manifold ids
+
+            if (current_manifold_id_of_cell != numbers::flat_manifold_id)
+              {
+                // set manifold of inner FACES of tets according to hex cell
+                // manifold id
+                for (const auto &face : table_3D_inner_faces)
```

Comment:
I *think* that this is the only place where you're actually using this variable. It might be easier to read if you just moved the variable declaration right into this block, before the `for` (or if you're bold, into `for (... : HERE)`). This way it's more clear what the variable represents because it is declared right next to the only place where it's used and we can see from the comment above the `for` loop and the context what the triplets are going to represent.

Reply:
I've changed the name of the variable, but I don't understand which variable declaration is meant.